tests/lib/rules/no-unnecessary-act.test.ts

```diff
@@ -0,0 +1,782 @@
+    {
+      code: `// case: RTU act wrapping non-RTL
+      import { act } from 'react-dom/test-utils'
+
+      test('valid case', async () => {
+        act(() => {
+          stuffThatDoesNotUseRTL();
+        });
+
+        await act(async () => {
+          stuffThatDoesNotUseRTL();
+        });
+
+        act(function() {
+          stuffThatDoesNotUseRTL();
+        });
+
+        act(function() {
+          return stuffThatDoesNotUseRTL();
+        });
+
+        act(() => stuffThatDoesNotUseRTL());
+      });
+      `,
+    },
+    {
+      settings: {
+        'testing-library/utils-module': 'test-utils',
+      },
+      code: `// case: RTL act wrapping non-RTL - AGR disabled
+      import { act } from '@testing-library/react'
+      import { waitFor } from 'somewhere-else'
+
+      test('valid case', async () => {
+        act(() => {
+          waitFor();
+        });
+
+        await act(async () => {
+          waitFor();
+        });
+
+        act(function() {
+          waitFor();
+        });
+
+        act(function() {
+          return waitFor();
+        });
+
+        act(() => waitFor());
+      });
+      `,
+    },
+    {
+      code: `// case: RTL act wrapping both RTL and non-RTL calls
+      import { act, render, waitFor } from '@testing-library/react'
+
+      test('valid case', async () => {
+        act(() => {
+          render(element);
+          stuffThatDoesNotUseRTL();
+        });
+
+        await act(async () => {
+          waitFor();
+          stuffThatDoesNotUseRTL();
+        });
+
+        act(function() {
+          waitFor();
+          stuffThatDoesNotUseRTL();
+        });
+      });
+      `,
+    },
+    {
+      settings: {
+        'testing-library/utils-module': 'test-utils',
+      },
+      code: `// case: non-RTL act wrapping RTL - AGR disabled
+      import { act } from 'somewhere-else'
+      import { waitFor } from '@testing-library/react'
+
+      test('valid case', async () => {
+        act(() => {
+          waitFor();
+        });
+
+        await act(async () => {
+          waitFor();
+        });
+
+        act(function() {
+          waitFor();
+        });
+
+        act(function() {
+          return waitFor();
+        });
+
+        act(() => waitFor());
+
+        act(() => {})
+        await act(async () => {})
+        act(function() {})
+      });
+      `,
+    },
+  ],
+  invalid: [
+    // cases for act related to React Testing Library
+    {
+      code: `// case: RTL act wrapping RTL calls - callbacks with body (BlockStatement)
+      import { act, fireEvent, screen, render, waitFor, waitForElementToBeRemoved } from '@testing-library/react'
+      import userEvent from '@testing-library/user-event'
+
+      test('invalid case', async () => {
+        act(() => {
+          fireEvent.click(el);
+        });
+
+        await act(async () => {
+          waitFor(() => {});
+        });
+
+        await act(async () => {
+          waitForElementToBeRemoved(el);
```

Comment:
Minor, but given the `async` identifier in the function these two should probably have the `await` (though it is not relevant for the test)

```suggestion
          await waitFor(() => {});
        });

        await act(async () => {
          await waitForElementToBeRemoved(el);
```

Reply:
As you mentioned, it's not relevant for the test so I don't tend to put attention to that 😅